### PR TITLE
demistomock - add functions docstrings

### PR DESCRIPTION
**Pull request by Itay4**
## Status
- [x] Ready

## Related Issues
https://github.com/demisto/etc/issues/26399

## Description
Added docstrings to funcs that later can be parsed to docs

### CODE REVIEW

**Comment by liorblob** [comment]
There are some missing functions, do we want to add them as part of this PR?

File: Tests/demistomock/demistomock.py [hunk 1]
```diff
@@ -385,23 +385,61 @@
...
+
+    :return: Arguments object
+    :rtype: ``dict``
+    """
     return {}
 
 
 def command():
+    """(Integration only)
+       Retrieves the integration command that is being ran
```
Comment (glicht) [inline Tests/demistomock/demistomock.py:409]:
```suggestion
       Retrieves the integration command that is being run
```

File: Tests/demistomock/demistomock.py [hunk 2]
```diff
@@ -385,23 +385,61 @@
...
+
+       :return: Integrations command name
+       :rtype: ``str``
+    """
     return ""
 
 
 def log(msg):
+    """
+    Prints a message to the server logs
```
Comment (glicht) [inline Tests/demistomock/demistomock.py:419]:
```suggestion
    Prints a message to the current incidents war room log
```

File: Tests/demistomock/demistomock.py [hunk 3]
```diff
@@ -412,61 +450,184 @@ def get(obj, field):
...
+
+    :return: The value of the extracted field
+    :rtype: ``str``
+    """
     return str(get(obj, field))
 
 
 def context():
+    """
+    Retrieves the context data object of the incident ran in
```
Comment (glicht) [inline Tests/demistomock/demistomock.py:470]:
```suggestion
    Retrieves the context data object of the current incident
```

File: Tests/demistomock/demistomock.py [hunk 4]
```diff
@@ -412,61 +450,184 @@ def get(obj, field):
...
+
+    :return: Context data object
+    :rtype: ``dict``
+    """
     return {}
 
 
 def uniqueFile():
+    """
+    Generate a random UUID
```
Comment (glicht) [inline Tests/demistomock/demistomock.py:480]:
```suggestion
    Generate a unique file name based upon a random UUID
```

File: Tests/demistomock/demistomock.py [hunk 5]
```diff
@@ -412,61 +450,184 @@ def get(obj, field):
...
 
 
 def error(msg, *args):
+    """
+    Prints a message to the server logs in error level
+
+    :type msg: ``str``
+    :param msg: The message to be logged
+
+    :type args: ``dict``
```
Comment (glicht) [inline Tests/demistomock/demistomock.py:534]:
```suggestion
    :type args: ``tuple``
```

File: Tests/demistomock/demistomock.py [hunk 6]
```diff
@@ -412,61 +450,184 @@ def get(obj, field):
...
 
 
 def debug(msg, *args):
+    """
+    Prints a message to the server logs in debug level
+
+    :type msg: ``str``
+    :param msg: The message to be logged
+
+    :type args: ``dict``
```
Comment (glicht) [inline Tests/demistomock/demistomock.py:551]:
```suggestion
    :type args: ``tuple``
```

File: Tests/demistomock/demistomock.py [hunk 7]
```diff
@@ -479,59 +640,99 @@ def executeCommand(command, args):
...
     """
     return integrationContext
 
 
 def incidents(incidents=None):
-    """
-    In Scripts this returns the `Incidents` list from the context
-
-    In integrations this is used to return incidents to the server
+    """(Integration only)
```
Comment (glicht) [inline Tests/demistomock/demistomock.py:727]: Not accurate. This function also exists in scripts. In scripts you get the incidents list from the context.

File: Tests/demistomock/demistomock.py [hunk 8]
```diff
@@ -412,61 +450,184 @@ def get(obj, field):
...
+    :rtype: ``None``
+    """
     if type(results) is dict and results.get("contents"):
         results = results.get("contents")
     log("demisto results: {}".format(json.dumps(results, indent=4, sort_keys=True)))
 
 
 def credentials(credentials):
+    """(Integration only)
+    Retrieves authentication parameter object
```
Comment (glicht) [inline Tests/demistomock/demistomock.py:587]:
```suggestion
    For integrations that support fetching credentials. Send the fetched credentials to the server.
```

File: Tests/demistomock/demistomock.py [hunk 9]
```diff
@@ -412,61 +450,184 @@ def get(obj, field):
...
+
+    :return: Credentials object
```
Comment (glicht) [inline Tests/demistomock/demistomock.py:589]: remove. There is nothing returned here. add just a param type.

File: Tests/demistomock/demistomock.py [hunk 10]
```diff
@@ -412,61 +450,184 @@ def get(obj, field):
...
+
+    :return: Object contains file ID, path and name
+    :rtype: ``dict``
+    """
     return {'id': id, 'path': 'test/test.txt', 'name': 'test.txt'}
 
 
 def investigation():
+    """
+    Retrieves the ID of the investigation in which being ran in
```
Comment (glicht) [inline Tests/demistomock/demistomock.py:610]:
```suggestion
    Retrieves the ID of the investigation in which being run in
```

File: Tests/demistomock/demistomock.py [hunk 11]
```diff
@@ -412,61 +450,184 @@ def get(obj, field):
...
 
 
 def executeCommand(command, args):
+    """(Script only)
+    Executes given integration command / script and arguments
+
+    :type command: ``str``
+    :param command: Integration command name or script name to run
+
+    :type args: ``args``
```
Comment (glicht) [inline Tests/demistomock/demistomock.py:625]:
```suggestion
    :type args: ``dict``
```

File: Tests/demistomock/demistomock.py [hunk 12]
```diff
@@ -479,59 +643,99 @@ def executeCommand(command, args):
...
+    :rtype: ``str``
+    """
     return params().get(param)
 
 
 def getArg(arg):
+    """
+    Extracts given argument from the arguments object
+
+    :type param: ``str``
```
Comment (glicht) [inline Tests/demistomock/demistomock.py:662]:
```suggestion
    :type arg: ``str``
```

File: Tests/demistomock/demistomock.py [hunk 13]
```diff
@@ -479,59 +643,99 @@ def executeCommand(command, args):
...
+    :param param: Argument to get value of
```
Comment (glicht) [inline Tests/demistomock/demistomock.py:663]:
```suggestion
    :param arg: Argument to get value of
```

**Pull request merged**